subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -76,7 +77,13 @@ private void notifyAfterEvaluate(ProjectEvaluationListener listener, ProjectInte
         } catch (Exception e) {
             if (state.hasFailure()) {
                 // Just log this failure, and pass the existing failure out in the project state
-                LOGGER.error("Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.", e);
+                boolean logStackTraces = project.getGradle().getStartParameter().getShowStacktrace() != ShowStacktrace.INTERNAL_EXCEPTIONS;
+                String infoMessage = "Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.";
```

Comment:
Yes that's right. It does not have to be a follow-up error in all cases.

How about something like:

`Project evaluation failed including an error in afterEvaluate {}. Run with --stacktrace for details of the afterEvaluate {} error.`